### ISSUE TITLE
Error in integration step for lab XRD data

### ISSUE THREAD

**Issue by dawkinsjer** [silx-kit/pyFAI#1689]
Hello everyone,

Sorry if I am not using proper etiquette, this is my first time posting on Github. I recently attended Jerome's workshop on using PyFAI at the ESRF virtual symposium in February. It was extremely helpful for me.

I am now trying to use PyFAI to integrate data from a lab diffractometer, using a Eiger2R_500K detector (75x75 pixel size, total size 469x975 pixels).

Every step is working perfectly (and GSAS-2 is not working at all, I think because the beam center is not in the image?). However I am getting an error at the Cake & Integration step. There is nothing that appears in the windows, so the integration is not being performed despite the fit in the previous step being excellent. Here is the error code I get:

ERROR:silx.gui.widgets.ThreadPoolPushButton:Error while executing callable pyFAI.gui.tasks.IntegrationTask.run.
Traceback (most recent call last):
  File "C:\Program Files\python310_venv\lib\site-packages\silx\gui\widgets\ThreadPoolPushButton.py", line 56, in run
    result = self.__callable(*self.__args, **self.__kwargs)
  File "C:\Program Files\python310_venv\lib\site-packages\pyFAI\gui\tasks\IntegrationTask.py", line 254, in run
    rings = unitutils.from2ThRad(rings, self.__radialUnit, self.__wavelength, self.__directDist)
  File "C:\Program Files\python310_venv\lib\site-packages\pyFAI\gui\utils\unitutils.py", line 76, in from2ThRad
    elif isinstance(twoTheta, collections.Iterable):
AttributeError: module 'collections' has no attribute 'Iterable'
ERROR:pyFAI-calib2.UNCAUGHT:'IntegrationProcess' object has no attribute '_IntegrationProcess__rings'
Traceback (most recent call last):
  File "C:\Program Files\python310_venv\lib\site-packages\pyFAI\gui\tasks\IntegrationTask.py", line 1006, in __integratingFinished
    self.__updateGUIWithIntegrationResult(self.__integrationProcess)
  File "C:\Program Files\python310
_venv\lib\site-packages\pyFAI\gui\tasks\IntegrationTask.py", line 1029, in __updateGUIWithIntegrationResult
    self._plot.setIntegrationProcess(integrationProcess)
  File "C:\Program Files\python310_venv\lib\site-packages\pyFAI\gui\tasks\IntegrationTask.py", line 772, in setIntegrationProcess
    self.__availableRings = integrationProcess.rings()
  File "C:\Program Files\python310_venv\lib\site-packages\pyFAI\gui\tasks\IntegrationTask.py", line 276, in rings
    return self.__rings
AttributeError: 'IntegrationProcess' object has no attribute '_IntegrationProcess__rings'

When searching for this error I found a post by Valentin Valls (https://github.com/silx-kit/pyFAI/issues/1178) which says that this can happen when the image and detector sizes are not the same. In my case I am manually configuring the detector, however the image and detector are both showing the same size, 469x975 pixels.

I was wondering if anyone had an idea what was going on? Here are some images:

![image](https://user-images.githubusercontent.com/105393844/167949559-400a4165-b895-4451-aeff-f444c13c73c7.png)
![image](https://user-images.githubusercontent.com/105393844/167949644-c14ea341-9d4b-4736-8b11-2c818b55f962.png)

I am attaching the .gfrm file in case it helps. I also tried from the .tiff, but got the same result.

[LaB6_standard.zip](https://github.com/silx-kit/pyFAI/files/8673523/LaB6_standard.zip)

Thank you very much for your time.

**Comment by kif** [comment]
Well, your image calibrates out of the box on my computer (linux, python3.9). 
Here is the content of the PONI-file:
``
poni_version: 2
Detector: Detector
Detector_config: {"pixel1": 7.5e-05, "pixel2": 7.5e-05, "max_shape": [469, 975]}
Distance: 0.13045236553399583
Poni1: 0.01570491430210719
Poni2: 0.04053664601151393
Rot1: -0.33907577432387515
Rot2: 0.013998616541804162
Rot3: 3.175534648892557e-08
Wavelength: 7.109999999999999e-11
```

I know some things changed in python3.10 and beside automatic tests, not much was tested with python3.10, and even less under windows. I will need to investigate.

**Comment by kif** [comment]
I think I spotted the bug ...

```In [3]: collections.Iterable
<ipython-input-3-6e2ce18f4e67>:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
  collections.Iterable
O```

**Comment by kif** [comment]
Can you confirm the last "master" version fixes the issue and reopen the issue if needed.

**Comment by dawkinsjer** [comment]
Hey, thanks so much for the help. I downloaded the new master file but get the same issue with Python3.10 on Windows. Once again the calibration is fine and is only at the Cake & Integration step that I get the aforementioned issue. Here is an image:

![image](https://user-images.githubusercontent.com/105393844/168128993-da7a949c-4d41-4994-bb7c-c7d1d955dd50.png)

However I created a Python3.9 venv and ran the new master, and everything works great. Integration is successful. For now I will run from Python3.9 to integrate my data.

![image](https://user-images.githubusercontent.com/105393844/168128233-ecce216f-dce3-496b-9707-50aee99e3b1d.png)

Thank you so much for your help. I hope this can help someone else who has the same issue running with Python3.10
